### DISCUSSION TITLE
PocketbaseArduino: A Pocketbase + Arduino Library

### DISCUSSION THREAD

**Discussion by jeoooo** [pocketbase/pocketbase#4126] [Show and tell]
# PocketbaseArduino: Pocketbase + Arduino Wrapper Library for ESP8266/ESP32

newbie dev here

starting off with the header file code for now...

GITHUB REPO: https://github.com/jeoooo/pocketbasearduino

**Comment by jeoooo** [comment]
# PocketbaseArduino 0.1.0 🚀 

https://github.com/jeoooo/PocketbaseArduino/releases/tag/0.1.0

implemented getOne() function for PocketbaseArduino, which allows me to return a specific record from the Pocketbase collection, tho still needs work for other functions and features for accessing Pocketbase backend.



![image](https://github.com/pocketbase/pocketbase/assets/53299170/bdaae7d5-ec52-459e-9539-ae185011fd85)

**Comment by ramadhanafif** [comment]
Cool project, I thought something similar in C, then wrap it with platform-specific method for esp boards

**Reply by jeoooo** [reply]
@ramadhanafif  

Thank you! I did initially want this to be written in C, but most tutorials i can find to make an Arduino library usually involves C++ 😅

**Comment by jeoooo** [comment]
# 🚀 PocketbaseArduino 0.3.1 is out!
![image](https://github.com/pocketbase/pocketbase/assets/53299170/e09c535a-156a-460b-8fd9-e236c3aa9016)

https://github.com/jeoooo/PocketbaseArduino/releases/tag/0.3.1



> ## Changes
> 
> - Add `deleteRecord()` function
> - Updated README!
> ```cpp
> record = pb.collection("notes").deleteRecord("50p6wpaeu4h5j9x");
> ```
> 
> 
> ## What's next?
> 
> - Improve documentation
> - add `create()` function
> - improve syntax for other body parameters (having to place nullptr works for now, **but is very inconvenient for developers**)
> ```cpp
> // current implementation
> String record = pb.collection("websites").getOne("1", "websites");
> 
> // "wtf i need to place nullptr just because its optional?" - some random dev
> String record = pb.collection("websites").getOne("1", nullptr);
> .... 
> 
> // and instead use
>  String record = pb.collection("websites").getOne("1").expand("websites");
> ```
> chain methods is makes it much more flexible in terms of how developers will write the syntax and using this library (hopefully 😀)
> 
> 
> **Wanna know what Pocketbase is?:** https://pocketbase.io/
> 
> 
> ## What's Changed
> * Add README by @jeoooo in https://github.com/jeoooo/PocketbaseArduino/pull/3
> * Add `deleteRecord()` function
> * Updated README!
> ```cpp
> record = pb.collection("notes").deleteRecord("50p6wpaeu4h5j9x");
> ```
> 
> 
> **Full Changelog**: https://github.com/jeoooo/PocketbaseArduino/compare/0.2.0...0.3.1
>

**Comment by RayPS** [comment]
Nice project, is this still being maintained?

**Reply by levydanqc** [reply]
It seems not.. Kinda sad because it was a nice idea! Would have loved to see authentification, realtime updates etc. implemented ! Maybe someone will pick it up

**Reply by RayPS** [reply]
Vibecoding should be fine as it only wraping up API requests.

**Reply by jeoooo** [reply]
Hi @RayPS  @levydanqc ! . I haven’t been able to maintain it for about 2 years since I’ve been busy with my main work, but I’m planning to return to it again later this year and continue development. 🙇‍♂️

Nice to hear you liked the idea! Things like authentication and realtime updates are on my list to look into once I can spend more time working on it again.

**Comment by jeoooo** [comment]
# 🚀 **PocketbaseExtended v1.0.0 is out!**

https://github.com/PocketBaseExtended/PocketbaseExtended/releases/tag/1.0.0

This is a complete rewrite of the library, previously known as `PocketbaseArduino`. Here's what's new:

## **🔐 Full Authentication System**
- `authWithPassword()`, `authRefresh()`, `setAuthToken()`, and `clearAuthToken()` are all supported now
- All requests automatically attach `Authorization: Bearer <token>` once authenticated

## **📦 Structured Responses via `PBResponse`**
- Every method now has an `Ex` counterpart (e.g. `getOneEx()`, `createEx()`) returning a `PBResponse` struct with `ok`, `statusCode`, `body`, and `error` fields.

## **✨ New Methods**
- `update()` / `updateEx()`
- `checkHealth()` 
- `getFileUrl()`

## **🔧 Other Improvements**
- `expand`, `fields`, and `skipTotal` query params now supported on `getOne` and `getList`
- Debug logging is now **off by default** (call `pb.setDebug(true)` to re-enable)
- Fixed a `getList` query parameter bug from v0.x

## **⚠️ Breaking Changes**
- Library is now `#include <PocketbaseExtended.h>` but a `PocketbaseArduino` typedef alias is kept so **existing v0.x sketches still compile without changes**
- Remove manual `<ESP8266HTTPClient.h>`, `<BearSSLHelpers.h>`, `<WiFiClientSecure.h>`. they're now handled internally
